cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/UserDbClient.java

```diff
@@ -0,0 +1,57 @@
+package no.unit.nva.cognito.service;
+
+import no.unit.nva.cognito.exception.UserServiceException;
+import no.unit.nva.database.DatabaseService;
+import no.unit.nva.useraccessmanagement.model.UserDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+public class UserDbClient implements UserApi {
+
+    private static final Logger logger = LoggerFactory.getLogger(UserDbClient.class);
+    public static final String ERROR_GETTING_USER = "Error getting user";
+    public static final String ERROR_CREATING_USER = "Error creating user";
+    public static final String ERROR_UPDATING_USER = "Error updating user";
+
+    private final DatabaseService databaseService;
+
+    public UserDbClient(DatabaseService databaseService) {
+        this.databaseService = databaseService;
+    }
+
+    @Override
+    public Optional<UserDto> getUser(String username) {
+        UserDto userDto = null;
+        try {
+            UserDto queryOject = UserDto.newBuilder().withUsername(username).build();
+            userDto = databaseService.getUser(queryOject);
+        } catch (Exception e) {
+            logger.error(ERROR_GETTING_USER, e);
+        }
+        return Optional.ofNullable(userDto);
+    }
+
+    @Override
+    public UserDto createUser(UserDto user) {
+        try {
+            databaseService.addUser(user);
+        } catch (Exception e) {
+            logger.error(ERROR_CREATING_USER, e);
+            throw new UserServiceException(e.getMessage(), e);
+        }
+        //TODO: maybe not necessary to return user here?
```

Comment:
@axthosarouris What do you think here?

Reply:
a) Is the resource being created in a way that entails that it exists before we return it?
b) Is it better to return the details that indicate where the resource is and an indication of when it is likely to be available?

Reply:
I am worried that the write to dynamodb might not be in effect at the time we read it back. If user object is not changed in any way as a part of the create operation I see no reason to return it.